tests/integrationtest/l4_crd_test.go

```diff
@@ -1476,6 +1476,256 @@ func TestCreateDeleteL4RuleSSLDefaultValues(t *testing.T) {
+	_, aviModel := objects.SharedAviGraphLister().Get(SINGLEPORTMODEL)
+	nodes := aviModel.(*avinodes.AviObjectGraph).GetAviVS()
+	g.Expect(nodes).To(gomega.HaveLen(1))
+	g.Expect(nodes[0].Name).To(gomega.Equal(fmt.Sprintf("cluster--%s-%s", NAMESPACE, SINGLEPORTSVC)))
+	g.Expect(nodes[0].Tenant).To(gomega.Equal(AVINAMESPACE))
+	g.Expect(nodes[0].PortProto[0].Port).To(gomega.Equal(int32(8080)))
+
+	// Check for the pools
+	g.Expect(nodes[0].PoolRefs).To(gomega.HaveLen(1))
+	address := "1.1.1.1"
+	g.Expect(nodes[0].PoolRefs[0].Servers[0].Ip.Addr).To(gomega.Equal(&address))
+
+	applicationProfileRef := proto.String("thisisaviref-l4-ssl-appprofile")
+	networkProfileRef := proto.String("thisisaviref-networkprofile-tcp-proxy")
+	sslKeyAndCertificateRefs := []string{"thisisaviref-sslkeyandcertref"}
+	sslProfileRef := proto.String("thisisaviref-sslprofileref")
+	// Create the L4Rule
+	SetupL4RuleSSL(t, L4RuleName, NAMESPACE, ports, applicationProfileRef, networkProfileRef, sslProfileRef, sslKeyAndCertificateRefs...)
+
+	g.Eventually(func() string {
+		l4Rule, _ := lib.AKOControlConfig().V1alpha2CRDClientset().AkoV1alpha2().L4Rules(NAMESPACE).Get(context.TODO(), L4RuleName, metav1.GetOptions{})
+		return l4Rule.Status.Status
+	}, 30*time.Second).Should(gomega.Equal("Accepted"))
+
+	// Apply the  L4Rule to Service
+	svcObj := (FakeService{
+		Name:         SINGLEPORTSVC,
+		Namespace:    NAMESPACE,
+		Type:         corev1.ServiceTypeLoadBalancer,
+		ServicePorts: []Serviceport{{PortName: "foo1", Protocol: "TCP", PortNumber: 8080, TargetPort: intstr.FromInt(8080)}},
+	}).Service()
+	svcObj.Annotations = map[string]string{lib.L4RuleAnnotation: L4RuleName}
+	svcObj.ResourceVersion = "2"
+	_, err := KubeClient.CoreV1().Services(NAMESPACE).Update(context.TODO(), svcObj, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("error in updating Service: %v", err)
+	}
+
+	expectedDefaultPoolName := fmt.Sprintf("cluster--%s-%s-%s-%d", NAMESPACE, SINGLEPORTSVC, "TCP", ports[0])
+	g.Eventually(func() bool {
+		found, aviModel := objects.SharedAviGraphLister().Get(SINGLEPORTMODEL)
+		if !found {
+			return false
+		}
+		nodes = aviModel.(*avinodes.AviObjectGraph).GetAviVS()
+		return g.Expect(nodes[0].AviVsNodeCommonFields).NotTo(gomega.BeZero()) &&
+			g.Expect(nodes[0].AviVsNodeGeneratedFields).NotTo(gomega.BeZero()) &&
+			g.Expect(nodes[0].PoolRefs[0].AviPoolCommonFields).NotTo(gomega.BeZero()) &&
+			g.Expect(nodes[0].PoolRefs[0].AviPoolGeneratedFields).NotTo(gomega.BeZero()) &&
+			g.Expect(nodes[0].L4PolicyRefs).To(gomega.HaveLen(0)) &&
+			g.Expect(nodes[0].DefaultPool).To(gomega.Equal(expectedDefaultPoolName))
+	}, 30*time.Second).Should(gomega.Equal(true))
+
+	l4Rule := FakeL4Rule{
+		Name:       L4RuleName,
+		Namespace:  NAMESPACE,
+		Ports:      ports,
+		SSLEnabled: true,
+	}
+	obj := l4Rule.L4Rule()
+	convertL4RuleToSSL(obj, ports, applicationProfileRef, networkProfileRef, sslProfileRef, sslKeyAndCertificateRefs...)
```

Comment:
@DixitAakash: Do we require line number 1561 to 1568? `SetupL4RuleSSL` has call to `convertL4RuleToSSL` so that should server our purpose. I am not getting use of this.

Reply:
**SetupL4RuleSSL** will actually create the L4rule which **convertL4RuleToSSL** will not. convertL4RuleToSSL is only used to get l4rule obj template which we compare with the actual object (ako graph layer obj) to check if CRD is applied.